13 Expanding System domain to all REST layers

Daniel-J-Mason opened the pull request:
Created a full set of layers from Controller to persistence for System.
System will grow in the future as remaining entities get added. 
Controller Methods will also grow to implement more searchable features in the future

Daniel-J-Mason committed:
System Infrastructure + Domain repository edit
Convert related domain records to domain classes.
Add domain repositories
Add a copy of Infrastructure utilities from Kafka
Add SystemMapper

Daniel-J-Mason committed:
Remove TODO note

Daniel-J-Mason committed:
Merge remote-tracking branch 'origin/development' into 13-rest-domain-repository-+-infrastructure-for-system

Daniel-J-Mason committed:
Update Mappers to include findAll() for filter types, and findByNameContains for all.

Add bean to enable Lazy loading by default to prevent excessive data loading

Pull development updates

Generate Persistence Repositories

Flagged liquibase systems table for update

Daniel-J-Mason committed:
Merge remote-tracking branch 'origin/development' into 13-rest-domain-repository-+-infrastructure-for-system

Daniel-J-Mason committed:
Remove insert/update/delete from repositories + mappers

Daniel-J-Mason committed:
Correct ILIKE search to %#{query}% <--- (DETERMINE IF THIS IS VALID IBATIS SYNTAX)

Daniel-J-Mason committed:
remove 0012-expand-systems-table.xml
remove 0014-add-star_class-to-systems-table.xml
change ALL tables to singular

Daniel-J-Mason committed:
remove 0012-expand-systems-table.xml
remove 0014-add-star_class-to-systems-table.xml
change ALL tables to singular

Daniel-J-Mason committed:
Fix commodity error from plurals
Update changelog to allow coords to be null
change x/y/z -> x_coordinate/y_coordinate/z_coordinate
Update processors to work with inbound schema changes
Remove star clas from nav route

pveeckhout committed:
13 - adding logback-spring.xml to projects

Daniel-J-Mason committed:
consolidate ed_system_address with eliteId

Daniel-J-Mason committed:
Merge remote-tracking branch 'origin/13-rest-domain-repository-+-infrastructure-for-system' into 13-rest-domain-repository-+-infrastructure-for-system

Daniel-J-Mason committed:
correct mapper to use ResultMap ILO results
create application layer for System
add System beans
add dev profile back to application.properties
temporarily remove logback from rest

Daniel-J-Mason committed:
remove unneeded files

Daniel-J-Mason committed:
add back logback-spring.xml, add modifications to root folder and UTC format
add logs file to gitignore

Commit:
13 - move all bean config to BeanConfig.java and fix wrongly configured Station beans

Commit:
Merge pull request #24 from ed-pilots-network/13-intermediate-review

13 - intermediate review

Daniel-J-Mason committed:
update naming convention for persistence repositories
replace get-named files for application with find-naming
fold in System bean configuration to BeanConfig
update logback default folder to prevent conflicts

Daniel-J-Mason committed:
Merge remote-tracking branch 'origin/development' into 13-rest-domain-repository-+-infrastructure-for-system

# Conflicts:
#	liquibase/src/main/resources/changelog/changelog-master.xml
#	message-processors/commodityv3-processor/src/main/java/io/edpn/backend/messageprocessor/commodityv3/infrastructure/persistence/mappers/CommodityEntityMapper.java
#	message-processors/commodityv3-processor/src/main/java/io/edpn/backend/messageprocessor/commodityv3/infrastructure/persistence/mappers/EconomyEntityMapper.java
#	message-processors/commodityv3-processor/src/main/java/io/edpn/backend/messageprocessor/commodityv3/infrastructure/persistence/mappers/HistoricStationCommodityMarketDatumEntityMapper.java
#	message-processors/commodityv3-processor/src/main/java/io/edpn/backend/messageprocessor/commodityv3/infrastructure/persistence/mappers/StationEntityMapper.java
#	message-processors/commodityv3-processor/src/main/java/io/edpn/backend/messageprocessor/commodityv3/infrastructure/persistence/mappers/StationStationTypeEntityMapper.java
#	message-processors/commodityv3-processor/src/main/java/io/edpn/backend/messageprocessor/commodityv3/infrastructure/persistence/mappers/StationTypeEntityMapper.java
#	message-processors/commodityv3-processor/src/main/java/io/edpn/backend/messageprocessor/commodityv3/infrastructure/persistence/mappers/SystemEntityMapper.java

pveeckhout commented on liquibase/src/main/resources/changelog/0001-create-system-table.xml:
```diff
@@ -0,0 +1,23 @@
+                <constraints nullable="false" unique="true"/>
+            </column>
+            <column name="elite_id" type="bigint"/>
+            <column name="x_coordinate" type="FLOAT8"/>
```
It is better to use `float` because it is a base supported liquibase type, and will make the changeset system agnoistic

pveeckhout commented on liquibase/src/main/resources/changelog/0022-expand-system-table.xml:
```diff
@@ -0,0 +1,40 @@
+
+    <changeSet id="b305dffc-efd6-4dd3-8b75-996b5ce0fe19" author="daniel-j-mason">
+        <addColumn tableName="system">
+            <column name="last_updated" type="TIMESTAMP WITHOUT TIME ZONE"/>
```
I would prefer to use `timestamp` since that is the liquibase type that will be translated to other systems.

additionally why would timezone not be needed / wanted here?

pveeckhout commented on liquibase/src/main/resources/changelog/0022-expand-system-table.xml:
```diff
@@ -0,0 +1,40 @@
+    <changeSet id="b305dffc-efd6-4dd3-8b75-996b5ce0fe19" author="daniel-j-mason">
+        <addColumn tableName="system">
+            <column name="last_updated" type="TIMESTAMP WITHOUT TIME ZONE"/>
+            <column name="needs_permit" type="BOOLEAN"/>
```
make type lowercase

pveeckhout commented on liquibase/src/main/resources/changelog/0022-expand-system-table.xml:
```diff
@@ -0,0 +1,40 @@
+        <addColumn tableName="system">
+            <column name="last_updated" type="TIMESTAMP WITHOUT TIME ZONE"/>
+            <column name="needs_permit" type="BOOLEAN"/>
+            <column name="population" type="BIGINT"/>
```
make type lowercase

pveeckhout commented on message-processors/commodityv3-processor/src/main/java/io/edpn/backend/messageprocessor/commodityv3/infrastructure/persistence/mappers/HistoricStationCommodityMarketDatumEntityMapper.java:
```diff
@@ -58,5 +80,5 @@ WHERE id IN (
                   AND (next_timestamp = timestamp + INTERVAL '1 microsecond' OR prev_timestamp = timestamp - INTERVAL '1 microsecond')
             )
             """)
-    int deleteObsoleteForStationAndCommodity(@Param("stationId") UUID stationId, @Param("commodityId") UUID commodityId);
```
this was changed in a later PR, what has this been reverted to the older naming?

pveeckhout commented on message-processors/commodityv3-processor/src/main/java/io/edpn/backend/messageprocessor/commodityv3/infrastructure/persistence/mappers/StationStationTypeEntityMapper.java:
```diff
@@ -0,0 +1,33 @@
+import java.util.UUID;
+
+@Mapper
+public interface StationStationTypeEntityMapper {
```
wasn't this whole jointable removed in a previous PR?

pveeckhout commented on message-processors/commodityv3-processor/src/main/java/io/edpn/backend/messageprocessor/commodityv3/infrastructure/persistence/mappers/StationEntityMapper.java:
```diff
@@ -16,22 +18,40 @@ public interface StationEntityMapper {
+            @Result(property = "economyEntityIdProportionMap", column = "id", many = @Many(select = "findEconomyProportionsByStationId")),
             @Result(property = "systemId", column = "system_id", javaType = UUID.class, typeHandler = UuidTypeHandler.class)
     })
-    @Select("SELECT id, name, ed_market_id, market_updated_at, has_commodities, system_id FROM stations WHERE id = #{stationId}")
```
stationType is missing as this should be a FK on Station to StationTypes

pveeckhout commented on message-processors/commodityv3-processor/src/main/java/io/edpn/backend/messageprocessor/commodityv3/infrastructure/persistence/mappers/SystemEntityMapper.java:
```diff
@@ -14,13 +15,28 @@ public interface SystemEntityMapper {
             @Result(property = "name", column = "name")
     })
-    @Select("SELECT id, name FROM systems WHERE id = #{id}")
+    @Select("SELECT id, name FROM system")
```
missing the coordinates

Daniel-J-Mason replied:
As far as I can tell commodityv3 doesnt track starpos so this wouldnt be used for this processor?
need confrimation on that

pveeckhout replied:
indeed, my bad. I was confused an thought this was in the Rest project

pveeckhout commented on message-processors/fsssignaldiscoveredv1-processor/src/main/java/io/edpn/backend/messageprocessor/fsssignaldiscoveredv1/infrastructure/persistence/mapper/SystemEntityMapper.java:
```diff
@@ -19,27 +19,27 @@ public interface SystemEntityMapper {
             @Result(property = "yCoordinate", column = "y_coordinate"),
             @Result(property = "zCoordinate", column = "z_coordinate")
     })
-    @Select("SELECT id, name FROM systems")
```
missing the coordinates

pveeckhout commented on rest/src/main/resources/application.properties:
```diff
@@ -1,6 +1,7 @@
 management.server.port=8080
 server.port=8080
 
+spring.profiles.active=dev
```
this does not belong in the main appilcation.properties. this can have unintended side effects.

either set via maven run time or environment variable. not via application.properties

Daniel-J-Mason replied:
Ive historically collapsed application.properties into one file that houses all of your runnable profiles.

But I can keep it going in separate files if we want to do something like application-dev.yaml

pveeckhout replied:
https://reflectoring.io/spring-boot-profiles/

--> it is much easier to find properties for a certain profile an to compare them between profiles if they are split in different files. Also,  the reference manual says that the multi-document syntax can lead to unexpected behavior.

They  way you were doing it here, was kind of contradictory

You are setting the active profile in the same multi-document file, which  defeats the purpose, as dev will always be active.

_you were correctly using the `spring.config.activate.on-profile=dev, but I removed that in a previous change._

the better way is to put all the shared (and static) properties in the application.properties, and then a file per environment.

running it with the correct profile:

`java -jar file.jar --spring-boot.run.profiles=production` or `mvn spring-boot:run -Dspring-boot.run.profiles=dev,pieter` to set the correct profile. 

this has the additional benefit of setting the correct profile for the logbck.xml

this means you can supply multiple profiles, where the later passed ones will overwrite properties of the previous.

you could make a `application-daniel.properties`, and run with profiles `prod,daniel` and you would run with prod properties, only overriding the once defined in  `application-daniel.properties`

Commit:
13 - fixing readded mapper endpoints, renaming all repo implementations to Mybatis and consolidating the initial database schema liquibase files

Daniel-J-Mason committed:
Fix CommodityEntityMapper and EconomyEntityMapper

Daniel-J-Mason committed:
Merge remote-tracking branch 'origin/13-merge-fix' into 13-rest-domain-repository-+-infrastructure-for-system

# Conflicts:
#	message-processors/commodityv3-processor/src/main/java/io/edpn/backend/messageprocessor/commodityv3/infrastructure/persistence/mappers/EconomyEntityMapper.java
#	message-processors/commodityv3-processor/src/main/java/io/edpn/backend/messageprocessor/commodityv3/infrastructure/persistence/mappers/StationStationTypeEntityMapper.java

Commit:
13 - version bumping message-processor lib dependency

Daniel-J-Mason committed:
Merge remote-tracking branch 'origin/13-merge-fix' into 13-rest-domain-repository-+-infrastructure-for-system

Daniel-J-Mason committed:
Merge pull request #26 from ed-pilots-network/13-merge-fix

13 - fixing readded mapper endpoints, renaming all repo implementatio…

Daniel-J-Mason committed:
Merge remote-tracking branch 'origin/13-rest-domain-repository-+-infrastructure-for-system' into 13-rest-domain-repository-+-infrastructure-for-system

Daniel-J-Mason committed:
Reorder liquibase to preven system station conflict
Update logback to prevent file conflict
Add stationTypeID to commodityv3
Remove StationStationTypeEntityMapper.java

Commit:
13 - reordering changelogs to prevent using FK of not yet existing tables

Commit:
13 - removing unneeded files
